addons/website/static/tests/builder/snippets_preprocess.test.js

```diff
@@ -33,3 +34,16 @@ test("preprocess modifies the snippet", async () => {
     await waitForEndOfOperation();
     expect(":iframe [data-snippet='s_banner']").toHaveClass("preprocessed");
 });
+
+test("Image carousel snippets should have unique IDs", async () => {
+    const { getEditableContent } = await setupWebsiteBuilderWithSnippet("s_image_gallery");
+
+    const ImageSnippetButtonSelector =
+        ".o-website-builder_sidebar  #snippet_groups .o_snippet[name='Images'] button";
+    await contains(ImageSnippetButtonSelector).click();
+    await confirmAddSnippet("s_image_gallery");
+
+    const imageCarousels = getEditableContent().querySelectorAll(".s_image_gallery .carousel");
+    expect(imageCarousels).toHaveCount(2);
+    expect(imageCarousels[0].id).not.toEqual(imageCarousels[1].id);
```

Comment:
Maybe check you don t have undefined value

Reply:
you can use     patchWithCleanup(uniqueId, { nextId: 0 });
to make your test determinist